libraries/joomla/database/driver/pgsql.php

```diff
@@ -1018,6 +1018,14 @@ public function updateObject($table, &$object, $key, $nulls = false)
 	 */
 	public function quoteBinary($data)
 	{
+		if (is_resource($data))
+		{
+			if (get_resource_type($data) === 'stream')
+			{
+				$data = stream_get_contents($data, -1, 0);
+			}
+		}
```

Comment:
```suggestion
		if (is_resource($data) && get_resource_type($data) === 'stream')
		{
			$data = stream_get_contents($data, -1, 0);
		}
```
Should work, too. The get_resource_type will only be called if $data is a resource.